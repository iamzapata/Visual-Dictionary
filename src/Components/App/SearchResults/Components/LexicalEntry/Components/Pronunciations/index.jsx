import React from "react"
import { arrayOf, shape, string } from "prop-types"
import PlayAudioButton from "Components/App/SearchResults/Components/PlayAudioButton"

const Pronunciations = ({ pronunciations }) => (
  <div className="Pronunciations">
    {pronunciations.map(p => (
      <span key={p.id}>
        /<span>{p.phoneticSpelling}</span>/
        <p>
          {p.dialects && (
            <span className="Dialects">
              {p.dialects.map(d => (
                <span key={d}>{d}</span>
              ))}
            </span>
          )}
          {p.audioFile && <PlayAudioButton audioFile={p.audioFile} />}
        </p>
      </span>
    ))}
  </div>
)

Pronunciations.propTypes = {
  pronunciations: arrayOf(
    shape({
      id: string.isRequired,
      phoneticSpelling: string.isRequired,
      dialects: arrayOf(string).isRequired,
      audioFile: string
    })
  ).isRequired
}

export default Pronunciations
