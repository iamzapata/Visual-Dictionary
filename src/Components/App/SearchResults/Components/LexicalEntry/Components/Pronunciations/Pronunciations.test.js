import React from "react"
import { render, cleanup } from "react-testing-library"
import Pronunciations from "./index"
import mockEntry from "Components/App/SearchResults/mocks"

describe("<Pronunciations />", () => {
  afterEach(cleanup)

  const createProps = () => {
    return [
      {
        id: "someStringId",
        phoneticSpelling: "Jiberish",
        dialects: ["High Valyrian"],
        audioFile: "myFile.file"
      }
    ]
  }

  it("Should render self", () => {
    const { container } = render(
      <Pronunciations pronunciations={createProps()} />
    )

    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toBeTruthy()
  })

  it("Should display phonetic spelling", () => {
    const { queryByText } = render(
      <Pronunciations pronunciations={createProps()} />
    )

    const phoneticSpelling = createProps()[0].phoneticSpelling

    const phoneticSpellingPresence = queryByText(phoneticSpelling)

    expect(phoneticSpellingPresence).toBeTruthy()
  })

  it("Should display dialects", () => {
    const { queryByText } = render(
      <Pronunciations pronunciations={createProps()} />
    )

    const dialects = createProps()[0].dialects[0]

    const dialectsPresence = queryByText(dialects)

    expect(dialectsPresence).toBeTruthy()
  })

  it("Should have a button to play audio files if audio files present", () => {
    const { container } = render(
      <Pronunciations pronunciations={createProps()} />
    )

    const pronunciationsWithAudioFile = mockEntry().pronunciations.filter(
      p => p.audioFile
    )
    const playAudioButton = container.firstChild.getElementsByTagName("button")

    expect(playAudioButton.length).toEqual(pronunciationsWithAudioFile.length)
    expect(playAudioButton).toBeTruthy()
  })
})
