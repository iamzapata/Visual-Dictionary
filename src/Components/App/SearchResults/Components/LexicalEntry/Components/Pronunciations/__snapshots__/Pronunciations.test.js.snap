// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pronunciations /> Should render self 1`] = `
<div
  class="Pronunciations"
>
  <span>
    /
    <span>
      Jiberish
    </span>
    /
    <p>
      <span
        class="Dialects"
      >
        <span>
          High Valyrian
        </span>
      </span>
      <button
        class="PlayAudioButton"
      >
        â–º
      </button>
    </p>
  </span>
</div>
`;
